flang/include/flang/ISO_Fortran_binding.h

```diff
@@ -155,6 +162,19 @@ typedef struct CFI_cdesc_t {
 #else
   CFI_dim_t dim[]; /* must appear last */
 #endif
+
+#ifdef __cplusplus
+  RT_API_ATTRS inline bool HasAddendum() const {
+    return extra & _CFI_ADDENDUM_FLAG;
+  }
+  RT_API_ATTRS inline void SetHasAddendum() { extra |= _CFI_ADDENDUM_FLAG; }
+  RT_API_ATTRS inline int GetAllocIdx() const {
+    return ((int)extra >> _CFI_ADDENDUM_BITS);
+  }
+  RT_API_ATTRS inline void SetAllocIdx(int pos) {
+    extra |= (pos << _CFI_ADDENDUM_BITS);
```

Comment:
This isn't going to work if the field is initially nonzero.  Define a mask macro for the field and use it above and also here.

Reply:
Mask added. Let me know if there is anything else.